Spigot-Server-Patches/0336-Extend-player-profile-API-to-support-skin-changes.patch

```diff
@@ -0,0 +1,177 @@
+             GameProfile result = server.getSessionService().fillProfileProperties(profile, true);
+             if (result != null) {
+                 this.profile = result;
+             }
+         }
+-        return profile.isComplete() && (!isOnlineMode || !textures || hasTextures());
++        return profile.isComplete() && (!(isOnlineMode || force) || !textures || hasTextures());
+     }
+ 
+     private static void copyProfileProperties(GameProfile source, GameProfile target) {
+diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
+index dd78a87b..e3f7be28 100644
+--- a/src/main/java/net/minecraft/server/WorldServer.java
++++ b/src/main/java/net/minecraft/server/WorldServer.java
+@@ -135,6 +135,12 @@ public class WorldServer extends World implements IAsyncTaskHandler {
+         return this;
+     }
+ 
++    // Paper start - Provide dimension number of world
++    public int getDimension() {
++        return dimension;
++    }
++    // Paper end
++
+     // CraftBukkit start
+     @Override
+     public TileEntity getTileEntity(BlockPosition pos) {
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+index 210e3bc4..b0335684 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+@@ -101,6 +101,12 @@ public class CraftWorld implements World {
+     }
+     // Paper end
+ 
++    // Paper start - Provide dimension number of world
++    public int getDimension() {
++        return world.dimension;
++    }
++    // Paper end
++
+     private static final Random rand = new Random();
+ 
+     public CraftWorld(WorldServer world, ChunkGenerator gen, Environment env) {
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index 6cbf429f..bd743be5 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -55,6 +55,7 @@ import org.bukkit.craftbukkit.CraftStatistic;
+ import org.bukkit.craftbukkit.CraftWorld;
+ import org.bukkit.craftbukkit.advancement.CraftAdvancement;
+ import org.bukkit.craftbukkit.advancement.CraftAdvancementProgress;
++import org.bukkit.craftbukkit.inventory.CraftItemStack;
+ import org.bukkit.craftbukkit.map.CraftMapView;
+ import org.bukkit.craftbukkit.map.RenderData;
+ import org.bukkit.craftbukkit.scoreboard.CraftScoreboard;
+@@ -1172,14 +1173,102 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+     public void setPlayerProfile(PlayerProfile profile) {
+         EntityPlayer self = getHandle();
+         self.setProfile(CraftPlayerProfile.asAuthlibCopy(profile));
+-        List<EntityPlayer> players = server.getServer().getPlayerList().players;
```

Comment:
As I said on last PR, this PR should only need to scope itself to fixing the player seeing its own skin.

Leave this change as it was, as it was fine for OTHER Entities. 
All the logic you wrote below for other entities is re-implementing the entire Entity Tracking system which we do not want to do.

So gut all that, keep this for statement, and make refreshPlayer() focused only on it self.